packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -0,0 +1,64 @@
+import { MutableRefObject, useEffect, useState } from 'react';
+import { useViewport } from '../../../hooks/useViewport';
+import { EChartsType } from 'echarts';
+import { ECHARTS_GESTURE } from '../../../common/constants';
+import { DEFAULT_DATA_ZOOM } from '../eChartsConstants';
+import { ViewportInMs } from '../types';
+import { Viewport } from '@iot-app-kit/core';
+
+// known bug with zooming into live data https://github.com/apache/echarts/issues/11679
+// this function will prevent the data from moving if we are zoomed in
+const onDataZoomEvent = (chart: EChartsType, setViewport: (viewport: Viewport, lastUpdatedBy?: string) => void) => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  const currentDataZoom = chart.getOption().dataZoom[0];
+
+  const newDataZoom = {};
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.start > 0 ? (newDataZoom.startValue = currentDataZoom.startValue) : (newDataZoom.start = 0);
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.end < 100 ? (newDataZoom.endValue = currentDataZoom.endValue) : (newDataZoom.end = 100);
+
+  setViewport(
+    { start: new Date(currentDataZoom.startValue), end: new Date(currentDataZoom.endValue) },
+    ECHARTS_GESTURE
+  );
+  chart.setOption(
+    {
+      dataZoom: newDataZoom,
+    },
+    { lazyUpdate: true }
+  );
+};
+
+export const useDataZoom = (chartRef: MutableRefObject<EChartsType | null>, viewportInMs: ViewportInMs) => {
+  const { lastUpdatedBy, setViewport } = useViewport();
+  const [isScrolling, setIsScroling] = useState(false);
+
+  // handle live mode
+  useEffect(() => {
+    const chart = chartRef.current;
+    if (chart && (!isScrolling || lastUpdatedBy === 'date-picker')) {
+      setIsScroling(false);
+      chart.setOption(
+        {
+          dataZoom: { ...DEFAULT_DATA_ZOOM, startValue: viewportInMs.initial, end: 100 },
+        },
+        { lazyUpdate: true }
+      );
+    }
+  }, [chartRef, viewportInMs, isScrolling]);
+
+  // create listener for DataZoom events so that all charts can sync up
+  useEffect(() => {
+    const chart = chartRef.current;
+    if (chart) {
+      chart.on('dataZoom', function () {
```

Comment:
you also need to unsubscribe from this 

return () => chart.off('dataZoom', handler)